### PR TITLE
Fix GLAD-related build problems on Linux

### LINKED ISSUE #69033: Build failure on Linux (NixOS) due to introduction of GLAD loader
### Godot version

4.0.dev (7f8ecffa56834dce3ccbd736738b613d51133dea)

### System information

Linux (NixOS), X11, OpenGL3

### Issue description

I'm getting a build error when trying to build the current `master` branch on NixOS:
```
modules/openxr/openxr_api.cpp:67:10: fatal error: GL/gl.h: No such file or directory
   67 | #include <GL/gl.h>
      |          ^~~~~~~~~
```
Link to the problematic include: https://github.com/godotengine/godot/blob/7f8ecffa56834dce3ccbd736738b613d51133dea/modules/openxr/openxr_api.cpp#L67

The problem was probably introduced by PR https://github.com/godotengine/godot/pull/68586, which replaces the GL dependency with the GLAD loader.

The PR removes the pkg-config lookup of the GL-headers:
https://github.com/godotengine/godot/pull/68586/files#diff-74d244f7adeae188c7346d9de7e0e41a13aed24e7f43c98ae7fe3ce2f5ccfc0dL359

GL headers are available system-wide on some Linux distributions (but not on NixOS) so I guess this didn't manifest while testing the PR?

### Steps to reproduce

I updated nixpkgs [godot_4](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/godot/4/default.nix)-derivation to use the latest godot commit.

I can provide detailed instructions if that actually helps (you would have to have an installation of `nix` or NixOS).

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by queezle42**
See discussion in https://github.com/godotengine/godot/issues/69033

This PR replaces the remaining includes of `<GL/gl.h>` with `"thirdparty/glad/glad/gl.h"` and adds missing extensions (`EXT_texture` and `EXT_texture_sRGB`, required by the OpenXR-module) to the GLAD loader.

Tested on Linux (NixOS); should only affect Linux builds.

*Bugsquad edit:*
- Fixes #69033

### CODE REVIEW

**Comment by akien-mga** [comment]
Looks good, but many of these `_EXT` defines are just the same as their non-`EXT` counterpart. I wonder if the problem isn't that we're using `_EXT` defines in places where what we need is already core?

**Comment by queezle42** [comment]
@akien-mga That works for everything, except `GL_RGB2_EXT` (from `EXT_texture`).

**Comment by queezle42** [comment]
I've removed `EXT_texture_sRGB` from GLAD and updated the commit to use non-`_EXT` defines where possible. `EXT_texture` is still required, unless there is a way to get rid of `GL_RGB2_EXT`.

**Comment by clayjohn** [comment]
> I've removed `EXT_texture_sRGB` from GLAD and updated the PR to use non-`_EXT` defines where possible. `EXT_texture` is still required, unless there is a way to get rid of `GL_RGB2_EXT`.

I think we can just get rid of ``GL_RGB2_EXT`` as it isn't exposed for Android or Windows. It was likely only included because it was available (cc @dsnopek) . I would combine the #ifdef WINDOWS section with the #else section and then get rid of the extensions in GLAD.

**Comment by queezle42** [comment]
@clayjohn I think that is correct.

Since that function (`get_swapchain_format_name`) only maps the format name but the format is not used anywhere else in the engine it can probably be removed without causing problems. I have updated the PR again.

File: modules/openxr/extensions/openxr_opengl_extension.cpp [hunk 1]
```diff
@@ -167,8 +167,8 @@ void OpenXROpenGLExtension::get_usable_swapchain_formats(Vector<int64_t> &p_usab
 	p_usable_swap_chains.push_back(GL_SRGB8_ALPHA8);
 	p_usable_swap_chains.push_back(GL_RGBA8);
 #else
-	p_usable_swap_chains.push_back(GL_SRGB8_ALPHA8_EXT);
-	p_usable_swap_chains.push_back(GL_RGBA8_EXT);
+	p_usable_swap_chains.push_back(GL_SRGB8_ALPHA8);
+	p_usable_swap_chains.push_back(GL_RGBA8);
```
Comment (akien-mga) [inline modules/openxr/extensions/openxr_opengl_extension.cpp:171]: The three `#if`/`#elif`/`#else` seem pretty redundant now since it's all the same code.

**Comment by queezle42** [comment]
I removed the libGL-related code path as suggested by @clayjohn and cleaned up the redundant ifdef.

**Comment by dsnopek** [comment]
> I think we can just get rid of GL_RGB2_EXT as it isn't exposed for Android or Windows. It was likely only included because it was available (cc @dsnopek)

Yeah, we don't actually need it, we just put all the available formats on that list so we can print out the list of formats that an OpenXR runtime reports as supported. It should be totally fine to drop it!

**Review by dsnopek — APPROVED** [review]
The code looks good to me! Assuming this compiles on both Linux and Windows (CI is running now), I think it should be good.

Thanks! :-)

**Comment by dsnopek** [comment]
Oh, but I just noticed that this has multiple commits. It'll need to get squashed down to a single commit before it can be merged:

https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by queezle42** [comment]
@dsnopek Done

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**